primer3/argdefaults.py

```diff
@@ -42,5 +54,308 @@ class Primer3PyArguments:
     temp_only: int = 0
     calc_type_wrapper = 'ANY'
 
+    dmso_conc: float = 0.0
+    dmso_fact: float = 0.6
+    formamide_conc: float = 0.8
+    annealing_temp_c: float = 50.0  # 70.0 or 120 in primer3 source
+
     def todict(self) -> dict:
         return dataclasses.asdict(self)
+
+
+TAGS_INTERVAL_LIST = set([
+    'SEQUENCE_INCLUDED_REGION',
+    'SEQUENCE_TARGET',
+    'SEQUENCE_EXCLUDED_REGION',
+    'SEQUENCE_INTERNAL_EXCLUDED_REGION',
+])
+
+TAGS_SIZE_RANGE_LIST = set(['PRIMER_PRODUCT_SIZE_RANGE'])
+
+# (semicolon separated list of integer "quadruples"; default empty)
+TAGS_SEMI_QUAD = set(['SEQUENCE_PRIMER_PAIR_OK_REGION_LIST'])
+
+TAGS_PATH_FIX = set([
+    'PRIMER_THERMODYNAMIC_PARAMETERS_PATH',
+    'PRIMER_MASK_KMERLIST_PATH',
+])
+
+# these tags can show up more than once in a boulder file.  We will append
+# them to a list
+TAGS_APPEND = set(['SEQUENCE_EXCLUDED_REGION'])
+
+Sequence_T = Union[List[Any], Tuple[Any, ...]]
```

Comment:
Is this upper snake case a convention that you want to follow for types in this codebase? (versus upper camelcase, e.g., `SequenceT`)

Reply:
I just some convention I adopted for aliasing longer type annotations based on old school `C` style typing for example `time_t`.  I can update it if you think upper camelcase is the way to go.

Reply:
Seems fine to me, we can just make sure we capture it in the style section of our development guide